test/LibraryManager.IntegrationTest/LibmanCompletionTests.cs

```diff
@@ -33,5 +32,152 @@ public void LibmanCompletion_Destination()
 
             LibmanTestsUtility.WaitForCompletionEntries(Editor, expectedCompletionEntries, caseInsensitive: true);
         }
+
+        [TestMethod]
+        public void LibmanCompletion_Provider()
+        {
+            _libManConfig.Open();
+
+            string[] expectedCompletionEntries = new[] {
+                "cdnjs",
+                "filesystem",
+                "unpkg",
+            };
+
+            Editor.Caret.MoveToBeginningOfFile();
+            Editor.Caret.MoveDown(4);
+            Editor.KeyboardCommands.Type("\"provider\":");
+
+            LibmanTestsUtility.WaitForCompletionEntries(Editor, expectedCompletionEntries, caseInsensitive: true);
+        }
+
+        [TestMethod]
+        public void LibmanCompletion_DefaultProvider()
+        {
+            _libManConfig.Open();
+
+            string[] expectedCompletionEntries = new[] {
+                "cdnjs",
+                "filesystem",
+                "unpkg",
+            };
+
+            Editor.Caret.MoveToBeginningOfFile();
+            Editor.Caret.MoveDown();
+            Editor.Caret.MoveToEndOfLine();
+            Editor.KeyboardCommands.Enter();
+            Editor.KeyboardCommands.Type("\"defaultProvider\":");
+
+            LibmanTestsUtility.WaitForCompletionEntries(Editor, expectedCompletionEntries, caseInsensitive: true);
+        }
+
+        [TestMethod]
+        public void LibmanCompletion_DefaultDestination()
+        {
+            _libManConfig.Open();
+            string[] expectedCompletionEntries = new[] {
+                "Properties/",
+                "wwwroot/",
+            };
+
+            Editor.Caret.MoveToBeginningOfFile();
+            Editor.Caret.MoveDown();
+            Editor.Caret.MoveToEndOfLine();
+            Editor.KeyboardCommands.Enter();
+            Editor.KeyboardCommands.Type("\"defaultDestination\":");
+
+            LibmanTestsUtility.WaitForCompletionEntries(Editor, expectedCompletionEntries, caseInsensitive: true);
+        }
+
+        [TestMethod]
+        public void LibmanCompletion_LibraryForCdnjs()
+        {
+            _libManConfig.Open();
+            string[] expectedCompletionEntries = new[] {
+                "bootstrap",
+                "jquery",
+            };
+
+            Editor.Caret.MoveToBeginningOfFile();
+            Editor.Caret.MoveDown(4);
+            Editor.KeyboardCommands.Type("\"provider\": \"cdnjs\",");
+            Editor.KeyboardCommands.Enter();
+
+            Editor.KeyboardCommands.Type("\"library\":");
+            LibmanTestsUtility.WaitForCompletionEntries(Editor, expectedCompletionEntries, caseInsensitive: true, timeout: 5000);
+
+            Editor.KeyboardCommands.Type("j");
```

Comment:
The rest of this test is just completion filtering.

Reply:
I see. I will remove the filtering stuff

---
In reply to: [193926074](https://github.com/aspnet/LibraryManager/pull/108#discussion_r193926074) [](ancestors = 193926074)